autoconf-archive: fix quoting of m4_fatal

corngood opened the pull request:
This fixes the `cava` package, and could potentially fix others.  I tested this as an override in `cava` and it worked, but I haven't built this in staging.

I believe it was broken by 59b91bae8c9f855411d9f4cb847a6d17e73284b5.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

corngood committed:
autoconf-archive: fix quoting of m4_fatal

This cherry-picks a patch set fixing calls to m4_fatal which caused some
packages to fail in autoconf (e.g. cava).

jopejoe1 commented:
Getting a hash mismatch

```
error: hash mismatch in fixed-output derivation '/nix/store/a0avdgs7l859rlxcb2z6y63a5kq5fsc4-ax_check_gl-fix.patch.drv':
        likely URL: https://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=patch;h=427e226a2fe3980388abffd6de25ed6b9591cce3
         specified: sha256-ajmyhz7boaRsM6wfLuKkZMo6pC53FwABX6p6ZcsoMV8=
            got:    sha256-gQbfXqfhpjjsUek6A4xqmnHdKPayhZhcXk6o4O+7Iqo=

```

corngood commented:
Ah, sorry, that was because I started with an override using `fetchpatch`, and changed it to `fetchurl` + explicit `name` here.  Only the name would have differed.  Fixed.

jopejoe1 reviewed (approved):
Tested building a few packages with this.

Scrumplex commented on pkgs/by-name/au/autoconf-archive/package.nix:
```diff
@@ -9,6 +9,15 @@ stdenv.mkDerivation rec {
 
+  patches = [
+    # fix https://bugs.gentoo.org/941845
+    (fetchurl {
```
what is the reason for using `fetchurl` here? `fetchpatch`/`fetchpatch2` should be preferred when fetching patches generated by git frontends

corngood replied:
I believe there's some bootstrap context where this package is evaluated, but `fetchpatch` is broken.  I think because the bootstrap `fetchurl` is incompatible with it?  It's easy to repro by evaluating this package for linux with the change.

Scrumplex replied:
If there would be a circular dependency with a fetcher, we generally include patches directly in Nixpkgs. Using `fetchurl` for patches means that the patch might become unreproducible if the upstream service changes

corngood replied:
I've updated this to include the patches in nixpkgs.  There were two other similar patches that were merged together, so I grabbed them all.

I think that's a good idea given these are small and it's a critical package.

corngood commented:
In gentoo they apparently just hacked `cava` to work-around this: https://bugs.gentoo.org/941845

Given how infrequently `autoconf-archive` is updated, and the possibility of this breaking other packages, I think it still might be a good idea to fix it here.

Hisui02 commented:
When is this going to be available in unstable branch?

vcunat commented:
My hope is that this weekend we can get it into 24.11 and soon afterwards into master (nixos-ustable, etc.)

paradoxical-dev commented:
Any update on the merge?

vcunat commented:
Yes, the stuff is in master and the related channels.
